sonata/main.py

```diff
@@ -1234,8 +1234,12 @@ def on_add_item(self, _widget, play_after=False):
             if self.current_tab == self.TAB_LIBRARY:
                 items = self.library.get_path_child_filenames(True)
                 self.mpd.command_list_ok_begin()
-                for item in items:
-                    self.mpd.add(item)
+                for item in items:     ######
+                     if item.endswith('.pls') or item.endswith('.PLS') or \
```

Comment:
There's no place which looks at the extension of the file to know if the file is a playlist or not, but MPD "knows" and already tells you if it is ([and you already used it before](https://github.com/multani/sonata/pull/85/files#diff-7b9aa4ce036b0612cc050d5b1fd01519R548))
But, using this information is probably non-trivial, so simply refactor these few lines of could would be enough, I guess.